devenv/lib/colima.py

```diff
@@ -0,0 +1,60 @@
+from __future__ import annotations
+
+import os
+import platform
+from shutil import which
+
+from devenv.constants import bin_root
+from devenv.constants import MACHINE
+from devenv.lib import archive
+
+_version = "0.6.2"
+_sha256 = {
+    "colima-Darwin-arm64": "add68719cc2a8e29be331177534db19812f9f79621425878cd88f890ca0db476",
+    "colima-Darwin-x86_64": "43ef3fc80a8347d51b8ec1706f9caf8863bd8727a6f7532caf1ccd20497d8485",
+    "colima-Linux-aarch64": "a04dbdfff0d913502b948ed1025d0d12485780493de24750b867c06a6325b5d5",
+    "colima-Linux-x86_64": "58e1a13fb0f693a6b28c2d935856e3f4bab6f0ab93f6bf4c860b5d59791ed04f",
+}
+
+
+class UnexpectedPlatformError(Exception):
+    pass
+
+
+def build_name() -> str | None:
+    system = platform.system()
+    if system == "Linux":
+        if MACHINE == "arm64":
+            return "colima-Linux-aarch64"
+        return f"colima-Linux-{MACHINE}"
+    elif system == "Darwin":
+        return f"colima-Darwin-{MACHINE}"
+    else:
+        raise UnexpectedPlatformError(f"Unexpected OS: {platform.platform()}")
+
+
+def download(name: str, into: str) -> None:
+    url = (
+        "https://github.com/abiosoft/colima/releases/download/"
+        f"v{_version}/{name}"
+    )
+
+    archive.download(url, _sha256[name], dest=f"{into}/colima")
+    os.chmod(f"{into}/colima", 0o775)
```

Comment:
one thing I noticed (and we need to fix this in general -- not just here) -- is a partially successful download will break the share data for ever

for instance:
- a partial download of the `colima` file
- a partially extracted archive
- a misconfigured python which pulls the wrong archive (will place the `3.x.x` directory that can't be cleaned up later)

we should probably:
1. download to a temporary location adjacent to the target
2. atomically move to the target

this is tricky to do for directories in a cross platform way unfortuantely -- for files it's easy (`os.replace` is idempotent / atomic)

Reply:
did it for files